src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgery.cs

```diff
@@ -170,7 +180,10 @@ private void ValidateTokens(HttpContext httpContext, AntiforgeryTokenSet antifor
                 deserializedRequestToken,
                 out message))
             {
-                throw new AntiforgeryValidationException(message);
+                var ex = new AntiforgeryValidationException(message);
+
+                _logger.ValidationFailed(ex);
+                throw ex;
             }
```

Comment:
Should this method log if it succeeds? See a log of begin / end pairs in other logging.